### PR TITLE
Added simple Java wrapper for double auction plus test classes

### PR DESCRIPTION
**Pull request by bherd-rb**
@davidrpugh This is the first instance of a Java-based auction wrapper. The runtime errors that I had (discussed in PR #5 disappeared when I integrated the wrapper into ESL itself. That indicates some sort of compiler incompatibility in my previous settings, the reason of which I haven't been able to figure out yet. 

Anyway -- it seems to work now. Thoughts and feedback are very much appreciated!

### CODE REVIEW

File: src/main/java/org/economicsl/auctions/singleunit/Service.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2017 Robert Bosch GmbH
```
Comment (bherd-rb) [inline src/main/java/org/economicsl/auctions/singleunit/Service.java]: @davidrpugh This file is just for testing purposes. We need to get rid of it before merging the stuff back.

**Comment by bherd-rb** [comment]
@davidrpugh I'll see if we can simplify the structure by using SBT instead of Maven also for the Java stuff.

**Comment by davidrpugh** [comment]
@bherd-rb I made a minor alteration to `build.sbt` to specify the compile order as `ScalaThenJava` and am getting the following errors...
```bash
$ sbt compile
[info] Loading global plugins from C:\Users\pughdr\.sbt\0.13\plugins
[info] Loading project definition from C:\Users\pughdr\Research\scalabm\auctions\project
[info] Set current project to auctions (in build file:/C:/Users/pughdr/Research/scalabm/auctions/)
[info] Compiling 33 Scala sources and 4 Java sources to C:\Users\pughdr\Research\scalabm\auctions\target\scala-2.12\classes...
[error] C:\Users\pughdr\Research\scalabm\auctions\src\main\java\org\economicsl\auctions\singleunit\JDoubleAuction.java:77:  error: incompatible types: PricingRule<T,Object> cannot be converted to PricingRule<T,Price>
[error]         Tuple2<Option<Stream<Fill<T>>>, DoubleAuction<T>> clear = auction.clear(p);
[error]                                                                                                                                   ^
[error] (compile:compileIncremental) javac returned nonzero exit code
[error] Total time: 16 s, completed Mar 22, 2017 4:04:37 PM

```

**Comment by bherd-rb** [comment]
@davidrpugh Yeah, I've just realised that there is still an issue with the clear function (haven't written any tests yet). I'll fix the problems and update the code.

**Comment by davidrpugh** [comment]
@bherd-rb Interestingly the making the following change...
```java
public Optional<ClearResult<T>> clear(PricingRule<T, Price> p) {
        ...
    }
```
...leads the Intellij IDE to say that this should lead to a compile error. However now if try to compile using SBT everything is OK.
```bash
$ sbt compile
[info] Loading global plugins from C:\Users\pughdr\.sbt\0.13\plugins
[info] Loading project definition from C:\Users\pughdr\Research\scalabm\auctions\project
[info] Set current project to auctions (in build file:/C:/Users/pughdr/Research/scalabm/auctions/)
[info] Compiling 33 Scala sources and 4 Java sources to C:\Users\pughdr\Research\scalabm\auctions\target\scala-2.12\classes...
[success] Total time: 19 s, completed Mar 22, 2017 4:13:23 PM
```

**Comment by bherd-rb** [comment]
@davidrpugh That's very strange indeed ... I was fiddling around with the second type parameter of the PricingRule for quite a while and I couldn't find any satisfying working solution. In order to get rid of the errors for the time being, I was just about to push a version where the pricing rule is already hardcoded in the clear function.

**Comment by bherd-rb** [comment]
@davidrpugh It really seems to be an IntelliJ-related issue. I can compile the latest version with both SBT and Maven without any problems, although IntelliJ complains about the parameters of the PricingRule. 

And the Maven-produced JAR file works well in my JADE-based test scenario (haven't tested the SBT one yet). No runtime issues.

**Comment by davidrpugh** [comment]
@bherd-rb Any last minute changes? If not then I am going to merge this branch.

**Comment by bherd-rb** [comment]
@davidrpugh I don't have any urgent changes at the moment. But I have to emphasise that I haven't really tested the implementation properly yet ... if you think that's OK, then feel free to merge.

**Comment by davidrpugh** [comment]
@bherd-rb This isn't being merged into `develop` yet; only into another feature branch that is still being worked on so I am not so worried about comprehensive testing.

**Pull request merged**